sdk/messaging/azeventhubs/internal/eh/eh_internal.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package eh
+
+// ConvertToInt64 converts any int-like value to be an int64.
+func ConvertToInt64(intValue any) (int64, bool) {
```

Comment:
Is the input to this always (or at least sometimes) `any`?  If not, you could use a generic implementation.
```go
func ToInt64[T int|int8|int16|int32|int64](v T) int64 {
	return int64(v)
}
```

Reply:
That would be nicer, actually. The value that gets passed in here comes from the AMQP message properties, so it ends up being typed 'any' since it comes across the wire.

Reply:
Ah bummer, makes sense though.

Reply:
Yeah, I felt the same way. I did use some generics in other spots though :)